pimd/pim_cmd_common.c

```diff
@@ -1829,11 +1829,19 @@ void pim_show_join(struct pim_instance *pim, struct vty *vty, pim_sgaddr *sg,
 
 static void pim_show_jp_agg_helper(struct vty *vty, struct interface *ifp,
 				   struct pim_neighbor *neigh,
-				   struct pim_upstream *up, int is_join)
+				   struct pim_upstream *up, int is_join,
+				   struct ttable *tt)
 {
-	vty_out(vty, "%-16s %-15pPAs %-15pPAs %-15pPAs %5s\n", ifp->name,
-		&neigh->source_addr, &up->sg.src, &up->sg.grp,
-		is_join ? "J" : "P");
+	char *table;
+
+	ttable_add_row(tt, "%s|%pPAs|%pPAs|%pPAs|%s", ifp->name,
+		       &neigh->source_addr, &up->sg.src, &up->sg.grp,
+		       is_join ? "J" : "P");
+
+	/* Dump the generated table. */
+	table = ttable_dump(tt, "\n");
```

Comment:
This should be done in the caller of this function, outside the for loop.

Reply:
Resolved, Thank you